server/app/Models/Doc.php

```diff
@@ -411,6 +341,17 @@ public function content()
         return $this->hasMany('App\Models\DocContent');
     }
 
+    public function fullContentHtml()
+    {
+        return $this->content()
+            ->orderBy('page')
+            ->get()
+            ->reduce(function ($fullContent, $content) {
```

Comment:
Same problem with `reduce` as with `filter` - this is to be avoided if at all possible, especially in cases like this where the content is *huge*.